test/kernels/test_qkernel.py

```diff
@@ -515,5 +516,136 @@ def test_user_parameters(self):
             self.assertEqual(list(qkclass.user_param_binds.keys()), qkclass.user_parameters)
 
 
+class TestQuantumKernelBatching(QiskitMachineLearningTestCase):
+    """Test QuantumKernel circuit batching
+
+    Creates a class factory that takes the QuantumInstance class,
+    applies a method decorator to the execute function and returns the modified class.
+    The decorator function records the actual number of circuits being passed
+    to execute does not exceed the batch_size requested by the Quantum Kernel.
+    Also checks that the sum of the batch sizes matches the total number of expected
+    circuits.
+    Checks batching with both statevector simulator and QASM simulator.
+    """
+
+    def count_circuits(self, func):
+        """Decorator to record the number of circuits passed to QuantumInstance.execute.
+
+        Args:
+            func (Callable): execute function to be wrapped
+
+        Returns:
+            Callable: function wrapper
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwds):
+            self.circuit_counts.append(len(args[1]))
+            return func(*args, **kwds)
+
+        return wrapper
+
+    def hook_quantum_instance(self):
+        """Hooks into QuantumInstance to monitor circuit batching.
+
+        Class factory that takes the QuantumInstance class, applies a method decorator to the
+        execute function and returns the modified class.
+        """
+
+        class HookedQuantumInstance(QuantumInstance):
```

Comment:
Do we really need a factory here? Can it be simplified?

Reply:
I absolutely agree that it seems overcomplicated, but I wasn't sure how to more simply hook into a method that we don't call explicitly. I'm probably missing something blindingly obvious. I'll revisit this.

Reply:
I was indeed missing the blindingly obvious. Now implemented without resorting to a factory.